### PR TITLE
Add typings built from JSDoc

### PR DESCRIPTION
**Pull request by Gerhut**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -3,6 +3,39 @@
...
+ * @property {boolean} [dots=false] Ensure trailing dots on FQDNs in content. Supports a limited amount of record types.
+ */
+
+/**
+ * @typedef {object} StringifyOptions
+ * @property {boolean} [sections=true] Whether to group records into sections.
+ * @property {boolean} [crlf=false] When `true`, emit `\r\n` instead of `\n` for the resulting zone file.
+ * @property {boolean} [dots=false] Ensure trailing dots on FQDNs in content. Supports a limited amount of record types. Default: `false`.
+ */
+
```
Comment (silverwind) [inline index.js]: Is there an option to keep those docstrings out of the source? I prefer not to have them there.

File: package.json [hunk 2]
```diff
@@ -6,6 +6,7 @@
   "repository": "silverwind/dnsz",
   "license": "BSD-2-Clause",
   "scripts": {
+    "build": "make build-typings",
```
Comment (silverwind) [inline package.json]: Better to add a `build` make target instead and have the `test` target depend on it. Thought I generally would prefer to not have a build step in first place and just have a static `.d.ts` file (which I can forget about until I change the API).

File: index.js [hunk 3]
```diff
@@ -309,3 +354,6 @@ module.exports.stringify = (data, {crlf = defaults.stringify.crlf, sections = de
 
   return `${output.trim()}${newline}`;
 };
+
+module.exports.parse = parse;
+module.exports.stringify = stringify;
```
Comment (silverwind) [inline index.js]: Are these necessary? Would prefer `const parse = module.exports.parse = () => {}` instead.

File: index.js (hunk 1 above)
Comment (Gerhut) [reply index.js]: These structure definition is reference in the following jsdocs, remove them will break the jsdoc type annotation.

File: package.json (hunk 2 above)
Comment (Gerhut) [reply package.json]: Removed the build scripts and make d.ts like manually written.
But I am keeping the typing test to verify the test script is running under the correct types.

File: index.js (hunk 3 above)
Comment (Gerhut) [reply index.js]: This is a limitation of typescript jsdoc parsing. Anyway, removed them all and make d.ts something like written manually.

**Comment by silverwind** [comment]
Thanks!

**Pull request merged**